pulp_container/constants.py

```diff
@@ -71,3 +75,34 @@
 SIGNATURE_PAYLOAD_MAX_SIZE = 4 * MEGABYTE
 
 SIGNATURE_API_EXTENSION_VERSION = 2
+
+MANIFEST_TYPE = SimpleNamespace(
+    IMAGE="image",
```

Comment:
Maybe a residue, but I see you asserting this value in the tests. Did not we agree on "unkown" for images of unknown types? What is the purpose of having the "image" type if all images are images by their nature?

Reply:
Regular image that adheres to the specs is friendly to the container runtimes, meaning that it can be instantiated by `podman run`, flatpak image, for example cannot be instantiated(podman run will fail), hence you have a special cli to manage flatpak images and not podman.
So ideally if the image fully adheres to the specs, meaning that manifest is of the v2 type, config and layers are regular ones then it's a regular application image. For the rest, if there are deviations like in the config type or layer type that is not part or the constants we have then it's an 'unkown' type of image (for us).
On the thing that all images are images, you are right. The question whether we want to distinguish them between those 2 cases i've described.

Reply:
I was thinking along the same lines as Ina. I thought the type `image` could be useful to differentiate "common" **container** images from other oci artifacts (like flatpak, or helm charts, etc).

Reply:
If the image fully adheres to the specs, it's media type is already shown in the Manifests endpoint as either "docker.manifest.v2" or "oci.manifest", correct? Clients do not read our API endpoints. So, this type thingy is irrelevant to them. However, the integrators can still benefit from the information we provide. For example, they can determine that if media_type is one of the above and type is set to "uknown", it can be treated as a regular image. How does that sound to you?

Reply:
Not sure what would be best approach. Concept of a manifest is a more low level thingy, and the fact that it splits into an image/artifact/list. End users usually are familiar with `container image concept`. So if we keep `list`, types we parse like (flatpak, bootable) then maybe it makes sense to leave `image` for the rest instead of `unknown`

Reply:
speaking of.. i did not really see this PR handling the `artifact`type, was it intentional?

Reply:
Okay, let's leave the `image` and `list` in place. We can still remove it later as needed. @git-hyagi, please, consider adding the `artifact` type to the list.